### PR TITLE
WIP Added fedora 32 image

### LINKED ISSUE #295: Upgrade to Python 3.8
Octomachinery requires Python 3.7+ as octomachinery relies on contextvars which doesn’t have a backport.
https://docs.octomachinery.dev/en/latest/getting-started.html#runtime-pre-requisites

### PR DESCRIPTION
**Pull request by saisankargochhayat**
## Related Issues and Dependencies
Fixes #295 
Added fedora 32 image. 

The issue with fasttext still occurs, I tried inside the venv with pip 9.0.1 we can install fast text, but pipenv install fails.

### CODE REVIEW

**Comment by fridex** [comment]
> The issue with fasttext still occurs, I tried inside the venv with pip 9.0.1 we can install fast text, but pipenv install fails.

What does `pip install pygit2` produce?

File: Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi8
+FROM fedora:32
```
Comment (goern) [inline Dockerfile:1]: please use a fully qualified registry, so that we know where the image came from.

**Review by goern — CHANGES_REQUESTED** [review]

**Comment by saisankargochhayat** [comment]
> > The issue with fasttext still occurs, I tried inside the venv with pip 9.0.1 we can install fast text, but pipenv install fails.
> 
> What does `pip install pygit2` produce?

```
(venv) bash-5.0$ pip --version
pip 9.0.1 from /tmp/venv/lib64/python3.6/site-packages (python 3.6)
(venv) bash-5.0$ pip install pygit2
Collecting pygit2
  Downloading https://files.pythonhosted.org/packages/20/02/25077cf7ac6599e0e6bd2c6836e7c7360244d2d7224d54e51218dbe00711/pygit2-1.3.0.tar.gz (243kB)
    100% |████████████████████████████████| 245kB 1.0MB/s 
Collecting cffi>=1.4.0 (from pygit2)
  Downloading https://files.pythonhosted.org/packages/50/ca/bbca0fd95b24a1d4f0d2e016f09f35ae68d4fe72bf34cc538d0a0d2d3e10/cffi-1.14.3-cp36-cp36m-manylinux1_x86_64.whl (400kB)
    100% |████████████████████████████████| 409kB 1.2MB/s 
Collecting cached-property (from pygit2)
  Downloading https://files.pythonhosted.org/packages/3b/86/85c1be2e8db9e13ef9a350aecd6dea292bd612fa288c2f40d035bb750ded/cached_property-1.5.1-py2.py3-none-any.whl
Collecting pycparser (from cffi>=1.4.0->pygit2)
  Downloading https://files.pythonhosted.org/packages/ae/e7/d9c3a176ca4b02024debf82342dab36efadfc5776f9c8db077e8f6e71821/pycparser-2.20-py2.py3-none-any.whl (112kB)
    100% |████████████████████████████████| 112kB 953kB/s 
Installing collected packages: pycparser, cffi, cached-property, pygit2
  Running setup.py install for pygit2 ... error
    Complete output from command /tmp/venv/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-1o4qie37/pygit2/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-c_8pgwea-record/install-record.txt --single-version-externally-managed --compile --install-headers /tmp/venv/include/site/python3.6/pygit2:
    running install
    running build
    running build_py
    creating build
    creating build/lib.linux-x86_64-3.6
    creating build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/submodule.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/remote.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/callbacks.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/config.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/settings.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/blame.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/errors.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/index.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/ffi.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/credentials.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/_run.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/refspec.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/__init__.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/utils.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/_build.py -> build/lib.linux-x86_64-3.6/pygit2
    copying pygit2/repository.py -> build/lib.linux-x86_64-3.6/pygit2
    creating build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/common.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/types.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/buffer.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/repository.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/strarray.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/oid.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/graph.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/blame.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/config.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/refspec.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/net.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/transport.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/remote.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/submodule.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/diff.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/merge.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/revert.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/clone.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/attr.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/errors.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/index.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/stash.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/indexer.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/checkout.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/pack.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/proxy.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/describe.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    copying pygit2/decl/callbacks.h -> build/lib.linux-x86_64-3.6/pygit2/decl
    running build_ext
    generating cffi module 'build/temp.linux-x86_64-3.6/pygit2._libgit2.c'
    creating build/temp.linux-x86_64-3.6
    building 'pygit2._pygit2' extension
    creating build/temp.linux-x86_64-3.6/src
    gcc -pthread -Wno-unused-result -Wsign-compare -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DNDEBUG -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -D_GNU_SOURCE -fPIC -fwrapv -fPIC -I/usr/local/include -I/tmp/venv/include -I/usr/include/python3.6m -c src/blob.c -o build/temp.linux-x86_64-3.6/src/blob.o
    In file included from src/blob.c:30:
    src/blob.h:33:10: fatal error: git2.h: No such file or directory
       33 | #include <git2.h>
          |          ^~~~~~~~
    compilation terminated.
    error: command 'gcc' failed with exit status 1
    
    ----------------------------------------
Command "/tmp/venv/bin/python3.6 -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-1o4qie37/pygit2/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-c_8pgwea-record/install-record.txt --single-version-externally-managed --compile --install-headers /tmp/venv/include/site/python3.6/pygit2" failed with error code 1 in /tmp/pip-build-1o4qie37/pygit2/
You are using pip version 9.0.1, however version 20.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

File: Dockerfile [hunk 2]
```diff
@@ -13,9 +13,10 @@ WORKDIR /home/user
 
 # Add the ssh key from local dir to container dir.
 # ADD github /home/user/.ssh/id_rsa
+# ADD known_hosts /home/user/.ssh/known_hosts
 
 RUN \
-    dnf install -y --setopt=tsflags=nodocs git python38 python3-pip gcc redhat-rpm-config python3-devel which gcc-c++ &&\
+    dnf install -y --setopt=tsflags=nodocs git libgit2 python38 python36 python3-pip gcc redhat-rpm-config python3-devel which gcc-c++ &&\
```
Comment (fridex) [inline Dockerfile:19]: Try with:

```suggestion
    dnf install -y --setopt=tsflags=nodocs git libgit2-devel python38 python36 python3-pip gcc redhat-rpm-config python3-devel which gcc-c++ &&\
```

Comment (saisankargochhayat) [reply Dockerfile:19]: Okay sure :+1:

Comment (saisankargochhayat) [reply Dockerfile:19]: pip 9.0.1 can install pygit2 inside the venv
however the pipenv run still fails for fasttext - 
```
  File "/home/user/.local/share/virtualenvs/user-5PivF_Od/lib/python3.6/site-packages/pipenv/patched/notpip/_internal/operations/prepare.py", line 294, in prepare_linked_requirement
    abstract_dist.prep_for_dist(finder, self.build_isolation)
  File "/home/user/.local/share/virtualenvs/user-5PivF_Od/lib/python3.6/site-packages/pipenv/patched/notpip/_internal/operations/prepare.py", line 127, in prep_for_dist
    self.req.run_egg_info()
  File "/home/user/.local/share/virtualenvs/user-5PivF_Od/lib/python3.6/site-packages/pipenv/patched/notpip/_internal/req/req_install.py", line 474, in run_egg_info
    command_desc='python setup.py egg_info')
  File "/home/user/.local/share/virtualenvs/user-5PivF_Od/lib/python3.6/site-packages/pipenv/patched/notpip/_internal/utils/misc.py", line 705, in call_subprocess
    % (command_desc, proc.returncode, cwd))
pipenv.patched.notpip._internal.exceptions.InstallationError: Command "python setup.py egg_info" failed with error code 1 in /tmp/tmpnlbed0tebuild/fasttext/

2020-09-21 18:10:45,988  43 WARNING  kebechet.managers.update.update:723: Failed to update dependencies to their latest version, reporting issue
2020-09-21 18:10:46,800  43 INFO     thoth.sourcemanagement.sourcemanagement:86: Issue already noted on upstream with title #Failed to update dependencies to their latest version
```

Comment (fridex) [reply Dockerfile]: Sadly, the log is not that helpful. Does pip install work in a virtual environment?

```
python3.6 -m venv venv
source venv/bin/activate
pip install --force-reinstall pip==9.0.1  # or replicate the virtual env used
pip install fasttext
```

Comment (saisankargochhayat) [reply Dockerfile]:
```
bash-5.0$ ls
CHANGELOG.md  MANIFEST.in  Pipfile	 README.rst  config  kebechet	   kebechet.egg-info  openshift  requirements.txt  test.py
Dockerfile    OWNERS	   Pipfile.lock  app.sh      github  kebechet-cli  known_hosts	      playbooks  setup.py	   workflows
bash-5.0$ cd /tmp
bash-5.0$ python3.6 -m venv venv
bash-5.0$ source venv/bin/activate
(venv) bash-5.0$ pip install --force-reinstall pip==9.0.1  # or replicate the virtual env used
Collecting pip==9.0.1
  Downloading https://files.pythonhosted.org/packages/b6/ac/7015eb97dc749283ffdec1c3a88ddb8ae03b8fad0f0e611408f196358da3/pip-9.0.1-py2.py3-none-any.whl (1.3MB)
     |████████████████████████████████| 1.3MB 1.1MB/s 
Installing collected packages: pip
  Found existing installation: pip 19.3.1
    Uninstalling pip-19.3.1:
      Successfully uninstalled pip-19.3.1
Successfully installed pip-9.0.1
(venv) bash-5.0$ pip install fasttext
Collecting fasttext
  Downloading https://files.pythonhosted.org/packages/f8/85/e2b368ab6d3528827b147fdb814f8189acc981a4bc2f99ab894650e05c40/fasttext-0.9.2.tar.gz (68kB)
    100% |████████████████████████████████| 71kB 955kB/s 
Requirement already satisfied: pybind11>=2.2 in ./venv/lib/python3.6/site-packages (from fasttext)
Requirement already satisfied: setuptools>=0.7.0 in ./venv/lib/python3.6/site-packages (from fasttext)
Collecting numpy (from fasttext)
  Downloading https://files.pythonhosted.org/packages/b8/e5/a64ef44a85397ba3c377f6be9c02f3cb3e18023f8c89850dd319e7945521/numpy-1.19.2-cp36-cp36m-manylinux1_x86_64.whl (13.4MB)
    100% |████████████████████████████████| 13.4MB 87kB/s 
Installing collected packages: numpy, fasttext
  Running setup.py install for fasttext ... done
Successfully installed fasttext-0.9.2 numpy-1.19.2
You are using pip version 9.0.1, however version 20.2.3 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.
```

Comment (saisankargochhayat) [reply Dockerfile]: fasttext does get install inside the venv without any issues

File: Dockerfile (hunk 1 above)
Comment (saisankargochhayat) [reply Dockerfile]: Thanks

File: Dockerfile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi8
+FROM docker.io/library/fedora:32
```
Comment (goern) [inline Dockerfile:1]: let's use `registry.fedoraproject.org` first ;)

**Review by goern — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 3 above)
Comment (saisankargochhayat) [reply Dockerfile]: Thanks, I have changed it.

**Comment by goern** [comment]
is this still valid?

**Comment by fridex** [comment]
> is this still valid?

I would consider this low priority - this was created to fix https://github.com/thoth-station/glyph/issues/66. If the internal trigger implementation would be in place, we would probably not need this change.

**Pull request closed without merging**